testsupport/tiers/nstemplateset.go

```diff
@@ -142,6 +143,34 @@ func UntilNSTemplateSetHasTemplateRefs(expectedRevisions TemplateRefs) wait.NSTe
 	}
 }
 
+// UntilNSTemplateSetHasStatusTemplateRefs checks if the NSTemplateTier has the expected template refs in the Status
+func UntilNSTemplateSetHasStatusTemplateRefs() wait.NSTemplateSetWaitCriterion {
+	return wait.NSTemplateSetWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.NSTemplateSet) bool {
+			// check that the status was updated with the expected template ref.
+			if (actual.Status.ClusterResources == nil && actual.Spec.ClusterResources != nil) ||
+				(actual.Status.ClusterResources != nil && actual.Spec.ClusterResources == nil) ||
+				actual.Status.ClusterResources.TemplateRef != actual.Spec.ClusterResources.TemplateRef {
+				return false
+			}
+			if !reflect.DeepEqual(actual.Status.Namespaces, actual.Spec.Namespaces) {
+				return false
+			}
+			// check expected feature toggles, if any
+			featureAnnotation, featureAnnotationFound := actual.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey]
+			if featureAnnotationFound {
+				if !reflect.DeepEqual(utils.SplitCommaSeparatedList(featureAnnotation), actual.Status.FeatureToggles) {
+					return false
+				}
+			}
```

Comment:
we should also take care of the case when the annotation is missing, correct?
```suggestion
			} else {
				if len(actual.Status.FeatureToggles) != 0 {
					return false
				}
			}
```

Reply:
If I'm not mistaken if the annotation is missing than we don't want to check the `actual.Status.FeatureToggles` field. Basically my understanding is that the annotation is optional we should not return false if it's not there. Only compare it with the status if it's set. 

Am I misunderstanding this ?

Reply:
you are completely right - the annotation is optional, and the status should reflect the existence/content of the annotation. This means that if the annotation is not set, then the featureToggles in the status should be empty.